stratify/_vinterp.pyx

```diff
@@ -502,6 +503,71 @@ def interpolate(z_target, z_src, fz_src, axis=-1, rising=None,
         modes. NaN extrapolation is the default.
 
     """
+    func = functools.partial(
+        _interpolate,
+        axis=axis,
+        rising=rising,
+        interpolation=interpolation,
+        extrapolation=extrapolation
+    )
+    if not hasattr(fz_src, 'compute'):
+        # Numpy array
+        return func(z_target, z_src, fz_src)
+
+    # Dask array
+    import dask.array as da
+
+    # Ensure z_target is an array.
+    if not isinstance(z_target, (np.ndarray, da.Array)):
+      z_target = np.array(z_target)
+
+    # Compute sensible output chunks
+    if axis < 0:
+      axis += fz_src.ndim
+    out_shape = list(fz_src.shape)
+    if z_target.ndim == 1:
+        out_shape[axis] = z_target.shape[0]
+    else:
+        out_shape[axis] = z_target.shape[axis]
+    out_chunks = [None if i == axis else 'auto' for i in range(fz_src.ndim)]
+    out_chunks = da.empty(
+        out_shape,
+        chunks=out_chunks,
+        dtype=fz_src.dtype,
+    ).chunks
+
+    # Ensure `fz_src` is not chunked along `axis`.
+    in_chunks = list(out_chunks)
+    in_chunks[axis] = fz_src.shape[axis]
+    fz_src = fz_src.rechunk(in_chunks)
```

Comment:
@bjlittle What do you think? Should we rechunk so the output chunks are a reasonable size? Or should we just ensure that the vertical dimension is not chunked and leave any rechunking to users of the library?

The downside of rechunking the input so the output has reasonable size chunks is that you will load more data into memory when the output has fewer vertical levels than the input.

Reply:
Simplified this to only rechunking the input array if strictly necessary.